File: backend/PUGS/TravelPlanningService/Models/TravelPlan.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Diagnostics;

namespace TravelPlanningService.Models
{
    public class TravelPlan
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        // Vlasnik plana - referenca na User.Id iz Auth servisa (bez FK jer je druga baza)
        [Required]
        public Guid OwnerId { get; set; }

        [Required, MaxLength(150)]
        public string Name { get; set; } = string.Empty;

        [MaxLength(1000)]
        public string? Description { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        [Range(0, double.MaxValue)]
        public decimal Budget { get; set; }

        [MaxLength(1000)]
        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigacione osobine
        public List<Destination> Destinations { get; set; } = new();
        public List<Activity> Activities { get; set; } = new();
        public List<ChecklistItem> ChecklistItems { get; set; } = new();
    }
}

File: frontend/src/models/TravelPlan.js
export class TravelPlan {
  constructor({ id, ownerId, name, description, startDate, endDate, budget, notes, createdAt, destinationsCount, activitiesCount }) {
    this.id = id;
    this.ownerId = ownerId;
    this.name = name;
    this.description = description;
    this.startDate = startDate;
    this.endDate = endDate;
    this.budget = budget;
    this.notes = notes;
    this.createdAt = createdAt;
    this.destinationsCount = destinationsCount ?? 0;
    this.activitiesCount = activitiesCount ?? 0;
  }

  get formattedDateRange() {
    const start = new Date(this.startDate).toLocaleDateString('sr-RS');
    const end = new Date(this.endDate).toLocaleDateString('sr-RS');
    return `${start} - ${end}`;
  }
}